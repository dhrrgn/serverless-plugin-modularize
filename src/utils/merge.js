const { mergeDeepWithKey } = require('ramda');

const mergeArrays = (paramName, val1, val2) => {
  if (Array.isArray(val1) && Array.isArray(val2)) {
    return [...val1, ...val2];
  }

  if (typeof val1 === 'object' && typeof val2 === 'object') {
    return Object.assign(val1, val2);
  }

  return val2 || val1;
};

module.exports = (mergedLeftObj, rightObj) =>
  mergeDeepWithKey(mergeArrays, mergedLeftObj, rightObj);
