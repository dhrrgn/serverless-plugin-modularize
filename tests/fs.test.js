import path from 'path';
import { resolve } from '../src/utils/fs';

const fileTests = [
  {
    label: 'loads js file',
    filename: './samples/example.js',
  },
  {
    label: 'loads json file',
    filename: './samples/example.json',
  },
  {
    label: 'loads yml file',
    filename: './samples/example.yml',
  },
  {
    label: 'loads yaml file',
    filename: './samples/example.yaml',
  },
];

describe('utils/fs', () => {
  fileTests.forEach(({ label, filename }) => {
    it(label, () => {
      expect(resolve(path.resolve(__dirname, filename))).toMatchSnapshot();
    });
  });

  it('throws invalid file', () => {
    const load = () => resolve(path.resolve(__dirname, './sample/example.txt'));
    expect(load).toThrow();
  });
});
