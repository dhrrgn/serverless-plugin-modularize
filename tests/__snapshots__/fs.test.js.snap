// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/fs loads js file 1`] = `
Object {
  "custom": Object {
    "exportedJsFile": true,
  },
}
`;

exports[`utils/fs loads json file 1`] = `
Object {
  "provider": Object {
    "environment": Object {
      "FOO": "BAR",
    },
  },
}
`;

exports[`utils/fs loads yaml file 1`] = `
Object {
  "resources": Object {
    "Resources": Object {
      "ExampleBucket": Object {
        "Properties": Object {
          "AccessControl": "PublicRead",
          "BucketName": "TestBucket",
        },
        "Type": "AWS::S3::Bucket",
      },
    },
  },
}
`;

exports[`utils/fs loads yml file 1`] = `
Object {
  "functions": Object {
    "test": Object {
      "handler": "src/fake.default",
    },
  },
}
`;
